### PR TITLE
iceberg-data, Row level Java API do not have any examples or descript…

### PR DESCRIPTION
**Pull request by GrigorievNick**
Fix for https://github.com/apache/iceberg/issues/3082
iceberg-data, Row-level Java API do not have any examples or descriptions on-site doc. 
So I add few very simple examples here.

### CODE REVIEW

File: site/docs/api.md [hunk 1]
```diff
@@ -70,6 +72,27 @@ Iterable<CombinedScanTask> tasks = scan.planTasks();
...
+To configure a scan, call `where` and `select` on the `ScanBuilder` to get a new `ScanBuilder` with those changes.
+
+```java
+scanBuilder.where(Expressions.equal("id", 5))
+```
+
+When a scan is configured, call method `build` to execute scan. `build` return `CloseableIterable<Record>`
+
+```java
+Iterable<Record> result = IcebergGenerics.read(table)
```
Comment (rdblue) [inline site/docs/api.md:92]: Nit: this should be `CloseableIterable`.

File: site/docs/api.md [hunk 2]
```diff
@@ -70,6 +72,27 @@ Iterable<CombinedScanTask> tasks = scan.planTasks();
...
+ScanBuilder scanBuilder = IcebergGenerics.read(table)
+```
+
+To configure a scan, call `where` and `select` on the `ScanBuilder` to get a new `ScanBuilder` with those changes.
+
+```java
+scanBuilder.where(Expressions.equal("id", 5))
+```
+
+When a scan is configured, call method `build` to execute scan. `build` return `CloseableIterable<Record>`
```
Comment (rdblue) [inline site/docs/api.md:89]: Should this mention that it is Iceberg's `Record` class?

Comment (GrigorievNick) [reply site/docs/api.md:89]: Yep, I will add details.

File: site/docs/api.md (hunk 1 above)
Comment (GrigorievNick) [reply site/docs/api.md]: fixed.

**Review by rdblue — APPROVED** [review]

**Comment by rdblue** [comment]
Thanks for adding this, @GrigorievNick! I merged it.

**Pull request merged**